test/built-ins/RegExp/prototype/unicodeSets/this-val-invalid-obj.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2022 Mathias Bynens, Ron Buckton, and the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-get-regexp.prototype.unicodeSets
+description: Invoked on an object without an [[OriginalFlags]] internal slot
+info: |
+    get RegExp.prototype.unicodeSets -> RegExpHasFlag
+
+    1. Let R be the this value.
+    2. If Type(R) is not Object, throw a TypeError exception.
+    3. If R does not have an [[OriginalFlags]] internal slot, then
+      a. If SameValue(R, %RegExpPrototype%) is true, return undefined.
+      b. Otherwise, throw a TypeError exception.
+features: [regexp-v-flag]
+---*/
+
+var unicodeSets = Object.getOwnPropertyDescriptor(RegExp.prototype, 'unicodeSets').get;
+
+assert.throws(TypeError, function() {
+  unicodeSets.call({});
+}, 'ordinary object');
+
+assert.throws(TypeError, function() {
+  unicodeSets.call([]);
+}, 'array exotic object');
+
+assert.throws(TypeError, function() {
+  unicodeSets.call(arguments);
+}, 'arguments object');
```

Comment:
I might suggest adding another test case for `unicodeSets.call(() => {})` in this test, its Type is Object even though its `typeof` is `"function"`.

Reply:
Note that these tests mirror the existing tests for other RegExp flags. I’ll go ahead and address your comment here, but as a follow-up we’ll want to update those other tests as well:

<img width="603" alt="image" src="https://user-images.githubusercontent.com/81942/180957761-e87e9921-206c-4242-bf7b-146cf9eaf095.png">

Same for your other comment below.

Reply:
Good point, I've done this in https://github.com/tc39/test262/pull/3615